### DISCUSSION TITLE
AlgoFi & Folks Finance methodology

### DISCUSSION THREAD

**Discussion by 0xngmi** [DefiLlama/DefiLlama-Adapters#4529] [General]
Lately, we've applied some methodology changes across AlgoFi and Folks Finance.

## Issues
 The issues with these protocols were that:
 - STBL, a stablecoin that can be minted like DAI by depositing collateral into AlgoFi, could be re-deposited into AlgoFi. If counted, this would mean that we'd be counting both the collateral backing it and the stablecoin, all within the same protocol.
 - AlgoFi cTokens (receipt tokens for their lending protocol) could be deposited into their DEX, where, if counted, would be double counted.
 - AlgoFi DEX LP tokens could be deposited into AlgoFi Lending, where again there's double counting

Furthermore, both AlgoFi and Folks Finance have a lot of staked ALGO tokens deposited within their protocols:
- Folks Finance built a liquid staking ALGO token, gALGO, of which 80% of supply is within Folks Finance's lending protocol
- AlgoFi allows people to lock ALGO and use that as collateral in AlgoFi (called vALGO)

This was quite non-standard since none of the other lending protocols we have on defillama are so dominated by staked assets, and furthermore gALGO and vALGO end up being both functionally very similar (since they are all mostly used as collateral in a lending protocol) but the way they are implemented means that gALGO is a liquid staking protocol while vALGO isn't.

## Conclusion
After considering everything, we decided to go with the following methodology, which aims to represent defi adoption on algorand as accurately as possible while also maintaining a methodology that is consistent with protocols in other chains:
- Remove all within-protocol double counting, which is the same methodology that we apply across all protocols in defillama
- Count both vALGO and gALGO within their respective lending protocols, since we count stETH on aave in Ethereum, and those tokens are deposited in a defi protocol and at risk
- Since gALGO implements a liquid staking protocol we'll list it as a standalone protocol, but as almost all of it's supply is deposited into other protocols, we'll mark it as double counted, meaning that it won't be counted into Folks Finance TVL by default (because we already count all the gALGO deposited into their lending protocol!)

We've created this discussion to explain our reasoning, be as transparent as possible about why we are making these changes and invite the algorand community to discuss them with us.

**Comment by 0xngmi** [comment]
*Comment removed because this was just a temporary bug that has since been fixed*

**Reply by benbio** [reply]
We think that the current version of displaying the Folks TVL in the chain is not correct.

We believe Folks Finance Algo Liquid Governance is not the same as Liquid Staking and should be considered its own category. Liquid staking on other systems is purely wrapping tokens to simplify access to certain reward dynamics. Indeed the Algo Liquid Governance provides alongside the rewards access to a set of features: 
- voting
- pre-minting
- early claim
- collateral 
- and more coming. 

In our opinion, the ideal solution to this is to simply count the ALGO deposited into Liquid Governance in our TVL and eliminate all other counts of gALGO on Folks Finance.

This would mean the ~$67M in Liquid Governance plus the ~$4M deposited to the lending markets would take Folks TVL to $71M. gALGO deposited to the lending markets can be ignored.

**Reply by JTInvestsInYou** [reply]
I agree with Benedetto here. It would make more sense to count the ALGO locked in governance that “created” the gALGO. That is the true TVL of gALGO. From there, it makes sense not to double count the gALGO on the lending platform. 

Benedetto’s change to the calculation makes the most sense. If Defillama would like to host a community Twitter Space with the Algorand community, I could help set that up! 👌

**Reply by 0xngmi** [reply]
> We believe Folks Finance Algo Liquid Governance is not the same as Liquid Staking and should be considered its own category

I understand there's some differences but id it really that different? lido stETH, is also available as collateral in multiple lending markets such as aave. Furthermore. bot projects wrap a chain-native system that provides rewards, allowing people to use assets within that system in defi applications while also earning rewards from the system

The big issue with your proposal is that it would break our methodology and give an advantage to algorand against all other chains. This is because for all other chains we don't count liquid staking TVL unless users turn on a toggle, but, if we implemented your proposal, for algorand we'd count TVL from a system very similar to liquid staking.

So the main issue is that we'd be giving algorand preferential treatment and our metrics would become biased,

**Reply by 0xngmi** [reply]
@JTInvestsInYou I'm happy to participate in any forum, but as mentioned before I can't participate in twitter spaces because it requires using your phone and that would prevent me from using a voice changer, thus I would have to doxx myself (i'm anon)

**Comment by gidonkatten** [comment]
> Folks Finance built a liquid staking ALGO token, gALGO, of which 80% of supply is within Folks Finance's lending protocol

At of the time of writing, there is 191,487,602 gALGO ciculating supply and 102,935,762 gALGO deposited which is 53.8% rounded up

**Reply by 0xngmi** [reply]
Yeah sorry about that, might have taken an outdated number

**Comment by fedchain-intern** [comment]
Gm. I agree with most of the points 0xngmi has mentioned and can appreciate that it will help to create a more accurate representation of DeFi adoption in Algorand. However, I am with Benedetto regarding counting the ALGO that is deposited into Liquid Governance as TVL and eliminating all other counts of gALGO on Folks Finance. I think it is important to note that gALGO minters are not just using their gALGO on the lending protocol. gALGO is also being outside of Folks Finance in other DeFi protocols, GameFi, NFT projects, etc. As such, limiting the TVL calculation of gALGO to the lending component would be **underestimating** the adoption of gALGO and hence DeFi on Algorand.

Please see attached link for a comprehensive list of gALGO use cases: https://folks.finance/galgoecosystem.html

**Reply by JTInvestsInYou** [reply]
💯

**Reply by 0xngmi** [reply]
stETH in ethereum is also being used in multiple DeFI protocols, but we still remove it from TVL unless users toggle it. So I don't fully understand this argument, if the same case is true for stETH and we don't do it, why would it be different for gALGO?

**Comment by Zenzakil** [comment]
I think that it would be more accurate to count Folks Finance TVL as the sum of all categories, i.e., Lending and Liquid Governance. Show the TVL of Folks Finance as the sum of these and make it possible to expand the information by showing the "sub-TVL" divided by categories. The total TVL shown on DefiLlama would be "transparent" (light gray) and the tooltip is going to say that it includes a share of "Liquid Governance" and therefore this is counted double. 

My opinion/proposal is based on multiple examples which confirm this mechanism is already used on DeFiLama, some of them are mentioned below: 
 
Bunny: Is a protocol which operates as yield aggregator, but also offers delegated staking on their own platform (i.e., Bunny Token pool). Its TVL is calculated as the sum of all categories and is shown in light gray and the tooltip. 
TVL shown is equal to $9,287,045
$1,020,500 deposited on the Bunny platform. The remaining are deposited on Pancake swap and Qubit, external protocols, thus making these count 2 times.

https://defillama.com/protocol/bunny

ANKR: Is a protocol which offers multiple features, (i.e., Liquid staking, delegated staking, liquidity providing etc.). Its TVL is calculated as the sum of all categories and is shown in light gray and the tooltip. 

 https://defillama.com/protocol/ankr

**Reply by 0xngmi** [reply]
The main issue here is that your proposal would lead to double-counting within Folks Finance. A large part of Liquid Governance TVL is in the lending protocol TVL, thus the simple solution of just summing both would lead to double-counting within Folks Finance, inflating it's TVL.

That doesn't seem to be the case in the examples you've mentioned, ankr is a liquid staking protocol and we count the TVL of just their staking derivatives, which leads to no overlap with other deposits/doublecounting within Ankr, and the same seems to be true for bunny. Please let me know if I'm wrong and I'll make any change necessary, for us it's very important to maintain a consistent methodology through our site.

What you seem to be pointing to is that TVL of both bunny and Ankr ends up being counted in other protocols, bunny because it deposits it's deposits somewhere else and ankr because other protocols can use their tokens. We prevent double-counting originating from these by removing the TVL of both of these projects from the chain TVL, so while we display them in our list, they are not counted into the sum of all TVL.

What's very important to understand here is that for neither of these protocols there's any double-counting going on internally, each deposit is only counted once. This is important since otherwise people could spin multiple re-hypothecation mechanisms within a protocol and inflate their TVL infinitely (we see this commonly in protocols that have an underlying protocol and then a farm where you can deposit your LP tokens for rewards, which then they try to count twice).

**Comment by 0xngmi** [comment]
Hey y'all, sorry for the delay in responses from us here, we've been quite busy with all the chaos created by FTX. Will start focusing on this now.

**Comment by 0xngmi** [comment]
Thinking that the solution that might work best in these cases is to keep chain TVL as is but change Folks Finance row to display their TVL without double counting.

atm we lack the infra for that but will be working on this

**Reply by mohammadfarari1360** [reply]
Okay good
